### DISCUSSION TITLE
Remediation script

### DISCUSSION THREAD

**Discussion by everartaraujo** [Devolutions/UniGetUI#2593] [Q&A] [answered]
Hello guys. 

I like this tool and would like to set it up for my small org. I use Intune and was thinking about deploying a remediation script to work on a schedule:

1: open UnigetUI as admin
2: check for updates
3: update everything without prompting the user to accept anything.
4: close Uniget UI

Is it possible at all? if not, can you guide me on what should be the best approach to solve this problem? I am a bit lost here and don't have a deep knowledge of PowerShell scripts but can find my way around.

**Comment by Lockszmith-GH** [comment, marked as answer]
You don't need UniGetUI - you can do that directly with [winget powershell's module](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/).
Or if you prefer chocolatey or scoop, use that instead.

What UniGetUI is doing is provide a UI to the underlying command line tools that you can already use.

Example code to collect the installed packages and list which are ready for updates:

```powershell
$wgupd = Get-WinGetPackage -OutVariable wglist | Where-Object { $_.IsUpdateAvailable }
Write-Host "Packages $($wglist.Count); Updates $($wgupd.Count)"
$wgupd | Sort-Object Name
```

And then update everything except packages that match either 'not-this-package' or 'bad-name'
```powershell
$wgupd | Where-Object Id -NotMatch 'not-this-package|bad-name' | Update-WinGetPackage -Verbose
```

**Reply by everartaraujo** [reply]
Thanks a lot, I will give a try. I had problems before running Winget in the system context that's why.

**Comment by marticliment** [comment]
@Lockszmith-GH is right. UniGetUI is not designed for automated tasks, WinGet is much more appropiate instead
